test/post_install_test/main.cpp

```diff
@@ -0,0 +1,79 @@
+#include "rodsClient.h"
+#include "connection_pool.hpp"
+#include "dstream.hpp"
+#include "transport/default_transport.hpp"
+#include "filesystem.hpp"
+
+#include <iostream>
+#include <string>
+
+namespace io = irods::experimental::io;
+namespace fs = irods::experimental::filesystem;
+
+int cleanup(irods::connection_pool::connection_proxy& _conn, const fs::path& _path);
+
+int main()
+{
+    rodsEnv env;
+    if (getRodsEnv(&env) != 0) {
+        std::cerr << "Error: cannot get iRODS environment\n";
+        return 1;
+    }
+
+    const std::string test_string = "This is a test!";
+    const auto data_obj_path = fs::path{env.rodsHome} / "post_install_test.txt";
+
+    irods::connection_pool conn_pool{1, env.rodsHost, env.rodsPort, env.rodsUserName, env.rodsZone, 600};
+    auto conn = conn_pool.get_connection();
+
+    // Create a new data object and write some bytes to it.
+    {
+        io::client::default_transport xport{conn};
+        io::odstream out{xport, data_obj_path};
+
+        if (!out) {
+            std::cerr << "Error: cannot open data object for writing [" << data_obj_path << "]\n";
+            return 1;
+        }
+
+        out << test_string;
+    }
+
+    // Check that the new data object contains the correct content.
```

Comment:
should also check the timestamp is fresh?   in case there was already one there from an earlier test run?

Reply:
Added timestamp to name.